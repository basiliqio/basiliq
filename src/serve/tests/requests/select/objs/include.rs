use super::*;

crate::run_test_request!(
    include_rel_m2o,
    Method::GET,
    "/public__peoples?include=public__favorite_color",
    200
);
crate::run_test_request!(
    include_rel_m2m,
    Method::GET,
    "/public__peoples?include=public__articles",
    200
);
crate::run_test_request!(
    include_rel_o2m,
    Method::GET,
    "/public__peoples?include=public__people-article",
    200
);

crate::run_test_request!(
    include_distant,
    Method::GET,
    "/public__peoples?include=public__articles.public__comments",
    200
);
crate::run_test_request!(
    include_distant_self,
    Method::GET,
    "/public__peoples?include=public__articles.public__comments.public__peoples",
    200
);
