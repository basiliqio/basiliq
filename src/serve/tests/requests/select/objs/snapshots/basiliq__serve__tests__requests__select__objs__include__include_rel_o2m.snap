---
source: src/serve/tests/requests/select/objs/include.rs
expression: res

---
{
  "jsonapi": {
    "version": "1.0"
  },
  "data": [
    {
      "type": "public__peoples",
      "id": "1649b1e9-8a5f-4f52-b331-c07ce3bccc6f",
      "attributes": {
        "age": 22,
        "first-name": "Francis",
        "gender": "M",
        "last-name": "Le Roy",
        "twitter": null
      },
      "relationships": {
        "public__people-article": {
          "data": [
            {
              "type": "public__people-article",
              "id": "1649b1e9-8a5f-4f52-b331-c07ce3bccc6f,fdf715dd-8772-498c-8196-6f4ccb64edef"
            },
            {
              "type": "public__people-article",
              "id": "1649b1e9-8a5f-4f52-b331-c07ce3bccc6f,2dbf5d1a-b029-4456-af6b-339c75b1089c"
            }
          ]
        }
      }
    },
    {
      "type": "public__peoples",
      "id": "777cc565-c66b-4942-ab44-8fc5f194b804",
      "attributes": {
        "age": 34,
        "first-name": "Somebody",
        "gender": "F",
        "last-name": "Wuhu",
        "twitter": "@randomhandle"
      },
      "relationships": {
        "public__people-article": {
          "data": [
            {
              "type": "public__people-article",
              "id": "777cc565-c66b-4942-ab44-8fc5f194b804,46c4fe50-8c56-4f26-935e-56ccfa496bb5"
            },
            {
              "type": "public__people-article",
              "id": "777cc565-c66b-4942-ab44-8fc5f194b804,fdf715dd-8772-498c-8196-6f4ccb64edef"
            }
          ]
        }
      }
    },
    {
      "type": "public__peoples",
      "id": "961e543a-4b22-4d48-a8e5-c1eafada950f",
      "attributes": {
        "age": null,
        "first-name": "AAAAAAAA",
        "gender": null,
        "last-name": "BBBBBBBBB",
        "twitter": null
      }
    }
  ],
  "included": [
    {
      "type": "public__people-article",
      "id": "1649b1e9-8a5f-4f52-b331-c07ce3bccc6f,2dbf5d1a-b029-4456-af6b-339c75b1089c"
    },
    {
      "type": "public__people-article",
      "id": "1649b1e9-8a5f-4f52-b331-c07ce3bccc6f,fdf715dd-8772-498c-8196-6f4ccb64edef"
    },
    {
      "type": "public__people-article",
      "id": "777cc565-c66b-4942-ab44-8fc5f194b804,46c4fe50-8c56-4f26-935e-56ccfa496bb5"
    },
    {
      "type": "public__people-article",
      "id": "777cc565-c66b-4942-ab44-8fc5f194b804,fdf715dd-8772-498c-8196-6f4ccb64edef"
    }
  ]
}
